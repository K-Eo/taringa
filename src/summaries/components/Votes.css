.Votes {
  border-right: 1px solid #dee2e6;
  padding-right: 0.5rem;
  text-align: center;
  display: inline-block;
  white-space: nowrap;
}

.item-big .Votes {
  border-right: none;
  display: inline-block;
  white-space: nowrap;
}

.Votes-Vote {
  display: inline-block;
  font-size: 80%;
  font-weight: 600;
  margin-bottom: 0;
  margin-top: 0;
  min-width: 38px;
}

.item-big .Votes-Vote,
.item-medium .Votes-Vote {
  display: block;
  min-width: 30px;
}

.item-small .Votes-Vote {
  min-width: 28px;
}

