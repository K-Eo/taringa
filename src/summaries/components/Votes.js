import React from 'react'

import './Votes.css'
import Action from '../../common/Action'

const Score = ({ upvotes, downvotes }) => {
  return (
    <div className="Votes">
      <Action icon="fa fa-arrow-up" />

      <div className="Votes-Vote">{(upvotes - downvotes).humanize()}</div>

      <Action icon="fa fa-arrow-down" />
    </div>
  )
}

Score.defaultProps = {
  downvotes: 0,
  upvotes: 0,
}

export default Score
