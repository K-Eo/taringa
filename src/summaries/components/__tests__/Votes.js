import React from 'react'

import Votes from '../Votes'

describe('Votes', () => {
  it('exists', () => {
    expect(Votes).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<Votes />)
    expect(wrapper).toMatchSnapshot()
  })

  it('calculates votes', () => {
    const wrapper = shallow(<Votes upvotes={10000} downvotes={2500} />)
    expect(wrapper).toMatchSnapshot()
  })
})
