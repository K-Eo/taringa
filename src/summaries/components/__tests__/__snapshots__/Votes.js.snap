// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Votes calculates votes 1`] = `
<div
  className="Votes"
>
  <Action
    icon="fa fa-arrow-up"
  />
  <div
    className="Votes-Vote"
  >
    7.5k
  </div>
  <Action
    icon="fa fa-arrow-down"
  />
</div>
`;

exports[`Votes renders default 1`] = `
<div
  className="Votes"
>
  <Action
    icon="fa fa-arrow-up"
  />
  <div
    className="Votes-Vote"
  >
    0
  </div>
  <Action
    icon="fa fa-arrow-down"
  />
</div>
`;
